manim/config/utils.py

```diff
@@ -70,201 +61,825 @@ def make_config_parser():
+        doc="A number between 0.0 (fully transparent) and 1.0 (fully opaque).",
+    )
 
-class JSONFormatter(logging.Formatter):
-    """Subclass of `:class:`logging.Formatter`, to build our own format of the logs (JSON)."""
+    frame_size = property(
+        lambda self: (self._d["pixel_width"], self._d["pixel_height"]),
+        lambda self, tup: (
+            self._d.__setitem__("pixel_width", tup[0])
+            or self._d.__setitem__("pixel_height", tup[1])
+        ),
+        doc="",
+    )
 
-    def format(self, record):
-        record_c = copy.deepcopy(record)
-        if record_c.args:
-            for arg in record_c.args:
-                record_c.args[arg] = "<>"
-        return json.dumps(
-            {
-                "levelname": record_c.levelname,
-                "module": record_c.module,
-                "message": super().format(record_c),
-            }
+    @property
+    def quality(self):
+        """Video quality."""
+        q = {
+            "pixel_width": self.pixel_width,
+            "pixel_height": self.pixel_height,
+            "frame_rate": self.frame_rate,
+        }
+        for qual in _QUAL:
+            if q == _QUAL[qual]:
+                return qual
+        else:
+            return None
+
+    @quality.setter
+    def quality(self, qual):
+        if qual not in _QUAL:
+            raise KeyError(f"quality must be one of {list(_QUAL.keys())}")
+        self.frame_size = _QUAL[qual]["pixel_width"], _QUAL[qual]["pixel_height"]
+        self.frame_rate = _QUAL[qual]["frame_rate"]
+
+    @property
+    def transparent(self):
+        """Whether the background opacity is 0.0."""
+        return self._d["background_opacity"] == 0.0
+
+    @transparent.setter
+    def transparent(self, val):
+        if val:
+            self.png_mode = "RGBA"
+            self.movie_file_extension = ".mov"
+            self.background_opacity = 0.0
+        else:
+            self.png_mode = "RGB"
+            self.movie_file_extension = ".mp4"
+            self.background_opacity = 1.0
+
+    @property
+    def dry_run(self):
+        """Whether dry run is enabled."""
+        return (
+            self.write_to_movie is False
+            and self.write_all is False
+            and self.save_last_frame is False
+            and self.save_pngs is False
+            and self.save_as_gif is False
         )
+
+    @dry_run.setter
+    def dry_run(self, val):
+        if val:
+            self.write_to_movie = False
+            self.write_all = False
+            self.save_last_frame = False
+            self.save_pngs = False
+            self.save_as_gif = False
+        else:
+            raise ValueError(
+                "It is unclear what it means to set dry_run to "
+                "False.  Instead, try setting each option "
+                "individually. (write_to_movie, write_alll, "
+                "save_last_frame, save_pngs, or save_as_gif)"
```

Comment:
**Comment** This sounds really fair. But I think it would be nice if dry_run is set back to false, then it revert to original state of `ManimConfig` object.

(see the post of the review for the revert thing)

Reply:
I thought of this, but the problem is *what is the "original state" of ManimConfig*? Is it the default state? Is it the state after you read some .cfg files? Is it the state just before you set dry_run to True?